autoload/go/guru.vim

```diff
@@ -217,11 +213,19 @@ endfunc
 
 " run_guru runs the given guru argument
 function! s:run_guru(args) abort
+
+  let old_gopath = $GOPATH
+  let $GOPATH = go#path#Detect()
   if go#util#has_job()
-    return s:async_guru(a:args)
+    let abort = s:async_guru(a:args)
```

Comment:
This has several meanings. For async it's the job id however for sync_guru it's the output. So the caller defines what to look. Can you rename it to `res` as for `result`? I think that would make more sense.